src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
```

Comment:
Using the reduce() method allows you to calculate the sum immediately without having to manually create the sum variable and use the forEach() loop. This is a more declarative approach that makes the code easier to understand and read.

<pre>
const sum = children.reduce((total, child) => {
  const mother = people.find((person) => person.name === child.mother);
  const ageDifference = child.born - mother.born;
  return total + ageDifference;
}, 0);
</pre>